kafka/source/pkg/adapter/adapter.go

```diff
@@ -134,6 +139,17 @@ func (a *Adapter) Start(ctx context.Context, stopCh <-chan struct{}) error {
 		kafkaConfig.Net.TLS.Config = tlsConfig
 	}
 
+	// Create the events pool
+	a.eventsPool = sync.Pool{
+		New: func() interface{} {
+			// The Pool's New function should generally only return pointer
+			// types, since a pointer can be put into the return interface
+			// value without an allocation:
+			val := cloudevents.New(cloudevents.CloudEventsVersionV02)
```

Comment:
this is likely something to be done for the channel too  ? 

(since both use the same common/handler for consumption)

Reply:
In channel probably the optimization could be different, i need to check it. Anyway, i think it's out of the scope of this pr